pkg/metrics/metrics.go

```diff
@@ -177,6 +185,21 @@ func CounterForSubscription(name, installedCSV string) prometheus.Counter {
 	return SubscriptionSyncCount.WithLabelValues(name, installedCSV)
 }
 
-func EmitCSVMetric(csv *olmv1alpha1.ClusterServiceVersion){
-	csvSyncCounter.WithLabelValues(csv.Name, csv.Spec.Version.String(), string(csv.Status.Phase), string(csv.Status.Reason)).Inc()
+func EmitCSVMetric(oldCSV *olmv1alpha1.ClusterServiceVersion, newCSV *olmv1alpha1.ClusterServiceVersion) {
+	if oldCSV == nil || newCSV == nil {
+		return
+	}
+
+	// Delete the old CSV metrics
+	csvAbnormal.DeleteLabelValues(oldCSV.Name, oldCSV.Spec.Version.String(), string(oldCSV.Status.Phase), string(oldCSV.Status.Reason))
```

Comment:
I think we need to include the namespace here so that we don't conflict if the same operator is installed multiple times.

I think we also want to avoid emitting for `copied` csvs, right?

Reply:
Good points, both are addressed in the latest version of the PR.